dspace-spring-rest/src/test/java/org/dspace/app/opensearch/OpenSearchControllerTest.java

```diff
@@ -0,0 +1,100 @@
+ * (Class has to start or end with IT to be picked up by the failsafe plugin)
+ *
+ * @author Oliver Goldschmidt (o dot goldschmidt at tuhh dot de)
+ */
+public class OpenSearchControllerTest extends AbstractControllerIntegrationTest {
+
+    @Test
+    public void searchAtomTest() throws Exception {
+        //When we call the root endpoint
+        getClient().perform(get("/opensearch/search")
+                                .param("query", "cats"))
+                   //The status has to be 200 OK
+                   .andExpect(status().isOk())
+                   //We expect the content type to be "application/atom+xml;charset=UTF-8"
+                   .andExpect(content().contentType("application/atom+xml;charset=UTF-8"))
+        ;
+    }
+
+    /* HTML is an open issue in OpenSearch, so skip this test at the moment
+    @Test
+    public void searchHtmlTest() throws Exception {
+        //When we call the root endpoint
+        getClient().perform(get("/opensearch/search")
+                                .param("query", "cats")
+                                .param("format", "html"))
+                   //The status has to be 200 OK
+                   .andExpect(status().isOk())
+                   //We expect the content type to be "application/atom+xml;charset=UTF-8"
+                   .andExpect(content().contentType("text/html;charset=UTF-8"))
+        ;
+    }
+    */
+
+    @Test
+    public void searchRssTest() throws Exception {
+        //When we call the root endpoint
+        getClient().perform(get("/opensearch/search")
+                                .param("query", "cats")
+                                .param("format", "rss"))
+                   //The status has to be 200 OK
+                   .andExpect(status().isOk())
+                   //We expect the content type to be "application/rss+xml;charset=UTF-8"
+                   .andExpect(content().contentType("application/rss+xml;charset=UTF-8"))
+        ;
+    }
+
+    @Test
+    public void serviceDocumentTest() throws Exception {
+        //When we call the root endpoint
+        getClient().perform(get("/opensearch/service"))
+                   //The status has to be 200 OK
+                   .andExpect(status().isOk())
+                   // and the contentType has to be an opensearchdescription
+                   .andExpect(content().contentType("application/opensearchdescription+xml;charset=UTF-8"))
+                   // and there need to be some values taken from the test configuration
+                   .andExpect(xpath("OpenSearchDescription/ShortName").string("DSpace"))
+                   .andExpect(xpath("OpenSearchDescription/LongName").string("DSpace at My University"))
+                   .andExpect(xpath("OpenSearchDescription/Description")
+                       .string("DSpace at My University DSpace repository")
+                   )
+        ;
+        /* Expected response for the service document is:
```

Comment:
Not sure if we have a general agreement on that. I would to suggest to move this comment in the javadoc of the test method but feel free to ignore my suggestion if nobody else agree (@tdonohue ? )

Reply:
Since this is in a Unit test I don't feel strongly here where it appears.  It might be slightly more accurate to have it in the JavaDocs, but I just like that it's documented somewhere.

Reply:
OK, I left it as it was :-)